connectors/sources/mongo.py

```diff
@@ -189,6 +190,15 @@ async def get_docs(self, filtering=None):
 
                 async for doc in self.collection.find(**find_kwargs):
                     yield self.serialize(doc), None
+
+            elif "aggregate" in advanced_rules:
+                aggregate_kwargs = deepcopy(advanced_rules.get("aggregate", {}))
```

Comment:
Why `deepcopy` here? Does the below code somehow modify the `aggregate_kwargs`?

Reply:
Yes see the line afterwards, also added the reason in the PR description.